homeassistant/components/mobile_app/http_api.py

```diff
@@ -75,6 +77,20 @@ async def post(self, request: Request, data: Dict) -> Response:
 
         data[CONF_USER_ID] = request["hass_user"].id
 
+        if slugify(data[ATTR_DEVICE_NAME], separator=""):
+            # if slug is not empty and would not only be underscores
+            # use DEVICE_NAME
+            pass
+        elif emoji.emoji_count(data[ATTR_DEVICE_NAME]):
```

Comment:
So, the problem is that slugify won't work if there is only emoji in here, right?

Why not by default demojize and slugy in a single step?

Reply:
Correct. 
I did consider that but if we have a device name that is something like "Fred's Phone🧛‍♀️🤘" We probably just want `freds_phone` which is what `slugify` gets us. Doing something like `slugify(emoji.demojize("Fred's Phone🧛‍♀️🤘"))` yields `'freds_phone_vampire_sign_of_the_horns' which is not really ideal especially as emojis give long names quickly.

Reply:
Agree 👍